Source/IGListCollectionViewLayout.mm

```diff
@@ -145,6 +147,7 @@ @implementation IGListCollectionViewLayout {
      -layoutAttributesForSupplementaryViewOfKind:atIndexPath:.
      */
     NSMutableDictionary<NSIndexPath *, UICollectionViewLayoutAttributes *> *_headerAttributesCache;
+    NSMutableDictionary<NSIndexPath *, UICollectionViewLayoutAttributes *> *_footerAttributesCache;
```

Comment:
Instead of adding another cache, how about extending the existing cache to be 2D:

```objc
NSMutableDictionary<NSString *, NSMutableDictionary *<NSIndexPath *, UICollectionViewLayoutAttributes *> *> *_supplementaryAttributesCache;
```

Initialized w/

```objc
_supplementaryAttributesCache = [NSMutableDictionary dictionaryWithDictionary:@[
    UICollectionElementKindSectionHeader: [NSMutableDictionary new],
    UICollectionElementKindSectionFooter: [NSMutableDictionary new],
  ];
```

That could even remove the need for `_supportedSectionElementKinds`, just iterate `allKeys` in the cache.